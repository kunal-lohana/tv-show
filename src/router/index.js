import Vue from "vue";
import VueRouter from "vue-router";
import TvShowList from "@/components/TvShowList/TvShowList.vue";
import ShowDetail from "@/components/ShowDetail/ShowDetail.vue";
import PageNotFound from "@/components/PageNotFound";

Vue.use(VueRouter);

export const routes = [
  {
    path: "/",
    component: TvShowList,
    meta: {
      title: "Tv Show"
    }
  },
  {
    path: "/:showId",
    component: ShowDetail,
    name: "showDetail",
    meta: {
      title: "Show Detail"
    }
  },
  {
    path: "*",
    name: "PageNotFound",
    component: PageNotFound,
    meta: {
      title: "Page-not-found"
    }
  }
];

export const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

router.afterEach(to => {
  document.title = to.meta.title || "Home";
});
