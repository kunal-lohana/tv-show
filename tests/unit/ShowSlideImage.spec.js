import {
  shallowMount,
  config,
  createLocalVue,
  RouterLinkStub
} from "@vue/test-utils";
import { imageData } from "./testingData";
import Vuetify from "vuetify";
import Vue from "vue";
import VueRouter from "vue-router";
import showSlideImage from "@/components/ShowSlideImage/ShowSlideImage.vue";
import { routes } from "@/router/index";

config.showDeprecationWarnings = false;
Vue.config.slient = false;

describe("Show Image Component ", () => {
  let localVue = createLocalVue();
  localVue.use(VueRouter);
  Vue.use(Vuetify);
  let vuetify;
  const router = new VueRouter({
    routes
  });
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(showSlideImage, {
      router,
      localVue,
      vuetify,
      propsData: {
        showImage: imageData
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("set prop. showImage data", async () => {
    const wrapper = shallowMount(showSlideImage, {
      router,
      localVue,
      vuetify
    });
    await wrapper.setProps({
      showImage: imageData
    });
    expect(wrapper.vm.showImage.length).toBe(imageData.length);
  });

  it("check showImage prop. data", () => {
    const wrapper = shallowMount(showSlideImage, {
      router,
      localVue,
      vuetify,
      propsData: {
        showImage: imageData
      }
    });
    expect(wrapper.vm.showImage.length).toBe(imageData.length);
  });

  it("check image url ", () => {
    const wrapper = shallowMount(showSlideImage, {
      router,
      localVue,
      vuetify,
      propsData: {
        showImage: imageData
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const imageUrl = wrapper
      .find('[data-test="show-slide-image"]')
      .attributes("src");
    expect(imageUrl).toEqual(imageData[0].image);
  });

  it("check router-link props", () => {
    const wrapper = shallowMount(showSlideImage, {
      router,
      localVue,
      vuetify,
      propsData: {
        showImage: imageData
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toEqual({
      name: "showDetail",
      params: { showId: 123 }
    });
  });
});
