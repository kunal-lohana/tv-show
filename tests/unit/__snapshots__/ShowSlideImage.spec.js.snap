// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Image Component  should match snapshot 1`] = `
<v-carousel-stub activeclass="v-window-item--active" mandatory="true" continuous="true" nexticon="$next" previcon="$prev" showarrows="true" showarrowsonhover="true" cycle="true" delimitericon="$delimiter" height="300" hidedelimiterbackground="true" interval="6000">
  <v-carousel-item-stub><a text="" color="secondary" class="text-decoration-none">
      <v-sheet-stub height="100%" tag="div">
        <v-row-stub tag="div" align="center" justify="center" class="fill-height">
          <v-img-stub options="[object Object]" position="center center" src="http://static.tvmaze.com/uploads/images/medium_portrait/1/4185.jpg" transition="fade-transition" data-test="show-slide-image" class="white--text align-end"></v-img-stub>
        </v-row-stub>
      </v-sheet-stub>
    </a></v-carousel-item-stub>
  <v-carousel-item-stub><a text="" color="secondary" class="text-decoration-none">
      <v-sheet-stub height="100%" tag="div">
        <v-row-stub tag="div" align="center" justify="center" class="fill-height">
          <v-img-stub options="[object Object]" position="center center" src="http://static.tvmaze.com/uploads/images/medium_portrait/1/4185.jpg" transition="fade-transition" data-test="show-slide-image" class="white--text align-end"></v-img-stub>
        </v-row-stub>
      </v-sheet-stub>
    </a></v-carousel-item-stub>
</v-carousel-stub>
`;
